joinservice/internal/watcher/validator.go

```diff
@@ -28,21 +29,23 @@ type Updatable struct {
 	mux         sync.Mutex
 	fileHandler file.Handler
 	variant     variant.Variant
+	cachedCerts *certcache.CachedCerts
 	atls.Validator
 }
 
 // NewValidator initializes a new updatable validator.
-func NewValidator(log *logger.Logger, variant variant.Variant, fileHandler file.Handler) (*Updatable, error) {
-	u := &Updatable{
+func NewValidator(log *logger.Logger, variant variant.Variant, fileHandler file.Handler) *Updatable {
+	return &Updatable{
 		log:         log,
 		fileHandler: fileHandler,
 		variant:     variant,
 	}
+}
 
-	if err := u.Update(); err != nil {
-		return nil, err
-	}
-	return u, nil
+// WithCachedCerts sets the available cached certificates in the updatable validator.
+func (u *Updatable) WithCachedCerts(cachedCerts *certcache.CachedCerts) *Updatable {
```

Comment:
What is the purpose of keeping this function?
We don't expect to ever call `NewValidator` in the join-service without it, so why not add `cachedCerts` as an argument to the constructor directly?

Reply:
Yes, this became obsolete with the latest update